### PR TITLE
Vectorize omnigenity objective over multiple surfaces

### PR DESCRIPTION
**Pull request by f0uriest**

### CODE REVIEW

File: tests/test_objective_funs.py [hunk 1]
```diff
@@ -1348,6 +1348,58 @@ def test_signed_plasma_vessel_distance(self):
...
+        obj1.build()
+        obj2.build()
+        obj3.build()
+        f1 = obj1.compute(*obj1.xs(eq, field))
+        f2 = obj2.compute(*obj2.xs(eq, field))
+        f3 = obj3.compute(*obj3.xs(eq, field))
+        # the order will be different but the values should be the same so we sort
+        # before comparing
+        np.testing.assert_allclose(
+            np.sort(np.concatenate([f1, f2])), np.sort(f3), atol=1e-14
```
Comment (YigitElma) [inline tests/test_objective_funs.py:1400]: why is the order different exactly?

Comment (f0uriest) [reply tests/test_objective_funs.py:1400]: its effectively flattening the (r,t,z) grid in different orders

**Comment by dpanici** [comment]
@dpanici change test tols for QFM here

**Comment by f0uriest** [comment]
why would the qfm test be failing here? nothing that changed should have any effect on that

**Review by YigitElma — APPROVED** [review]

**Comment by dpanici** [comment]
No idea, I will see and possibly adjust the test today

**Comment by YigitElma** [comment]
I don't know why it was failing on this PR but I guess it was due to some numerical thing with CI, and not an actual problem. The test passes now.

**Review by ddudt — APPROVED** [review]

**Pull request merged**